queries.sql

```diff
@@ -1,46 +1,22 @@
-/*TRANSACTIONS*/
-BEGIN;
-UPDATE animals SET species = 'unspecified';
-ROLLBACK;
-SELECT * FROM animals;
+/* Write query using JOIN */
 
-BEGIN;
-UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
-UPDATE animals SET species = 'pokemon' WHERE species is NULL;
-COMMIT;
-SELECT * FROM animals;
+/*What animals belong to Melody Pond*/
+SELECT name FROM animals INNER JOIN owners ON animals.owners_id = owners.owners_id WHERE owners.full_name = 'Melody Pond';
 
-BEGIN;
-DROP TABLE animals;
-ROLLBACK;
-COMMIT;
-SELECT * FROM animals;
+/*List of all animals that are pokemon*/
+SELECT animals.name FROM animals JOIN species ON animals.species_id = species.species_id WHERE species.name = 'Pokemon';
 
-BEGIN;
-DELETE FROM animals WHERE date_of_birth > '2022-01-01';
-SELECT * FROM animals;
-SAVEPOINT deleted_animals;
-UPDATE animals SET weight_kg = weight_kg * -1;
-ROLLBACK TO deleted_animals;
-SELECT * FROM animals;
-UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
-COMMIT;
-SELECT * FROM animals;
+/* List all owners and their animals, remember to include those that don't own any animal.*/
+SELECT owners.full_name, animals.name FROM animals RIGHT JOIN owners ON animals.owners_id = owners.owners_id ;
 
-/*Use aggregate function Count to count the number of animals*/
-SELECT COUNT(*) from animals;
+/* How many animals are there per species?*/
+SELECT species.name, COUNT(*) FROM animals JOIN species ON animals.species_id = species.species_id GROUP by species.name;
 
-/*Count the number of animals that had not made attempt to escape*/
-SELECT COUNT(*) from animals WHERE escape_attempts = 0;
+/*List all Digimon owned by Jennifer Orwell*/
+SELECT animals.name FROM animals JOIN owners ON animals.owners_id = owners.owners_id JOIN species ON animals.species_id = species.species_id WHERE owners.full_name = 'Jennifer Orwell' AND species.name = 'Digimon';
 
-/* Use aggregate function Average to determine the average weight of all the animals*/
-SELECT ROUND(AVG(weight_kg)::numeric, 3) FROM animals;
+/* List all animals owned by Dean Winchester that haven't tried to escape.*/
+SELECT animals.name FROM animals JOIN owners ON animals.owners_id = owners.owners_id WHERE owners.full_name = 'Dean Winchester' AND animals.escape_attempts = 0 GROUP BY animals.name;
```

Comment:
- [ OPTIONAL ] Great job so far but I think here you don't need to make group by animals.name because the question is list all animals owned by Dean Winchester so you already added to your condition where owners.full_name = 'Dean Winchester ' so I think you don't need the group by condition 💯😊🚀🚀